### PR TITLE
Added firebase to the repo

### LINKED ISSUE #48: Add general firebase library. (Please add latest stable version of firebase)

### PR DESCRIPTION
**Pull request by SethY8s**
Integrated Firebase into the repository. I’ve used the config from V1 with a few minor adjustments, and we’ll reintroduce some features at a later stage. This PR primarily ensures that we can successfully query data from Firebase and that environment variables (ENVs) are functioning correctly.

While the current code may appear a bit messy, it serves as the foundation for future Firebase queries and development. If you'd prefer a cleaner codebase before each PR, please let me know. 

Firebase Config features that need to be added back.
-typed collection groups using zod to typecast the different data coming back from the DB.
You can see that start of this at the bottom of the DB, but the code is commented out. After Zod is added we can start this typecasting process and determine what the structure of this DB will look like.

### CODE REVIEW

File: app/(tabs)/explore.tsx [hunk 1]
```diff
@@ -1,29 +1,61 @@
...
+import { ExternalLink } from "~/components/ExternalLink";
+import ParallaxScrollView from "~/components/ParallaxScrollView";
+import { ThemedText } from "~/components/ThemedText";
+import { ThemedView } from "~/components/ThemedView";
+
+import { collection, getDocs } from "firebase/firestore";
+import { db } from "~/firebase/config";
 
 export default function TabTwoScreen() {
+  const [testData, setTestData] = useState<any>([]);
```
Comment (kekoawong) [inline app/(tabs)/explore.tsx:15]: Nit - would prefer these typed but we can fix later

File: firebase/config.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * @template T
+ * @param {string} collectionName
+ * @return {*}  {CollectionReference<T>}
+ */
+const createCollection = <T = DocumentData>(collectionName: string): CollectionReference<T> => {
+    const converter = genericConverter<T>() as FirestoreDataConverter<T>;
+    return collection(db, collectionName).withConverter<T>(converter);
+};
+
```
Comment (kekoawong) [inline firebase/config.ts:76]: nice carrying this code over

File: app/(tabs)/explore.tsx [hunk 3]
```diff
@@ -32,33 +64,48 @@ export default function TabTwoScreen() {
...
+      <View>
+        <Text className="text-red-400">testing</Text>
+      </View>
+      {testData.map((item: any) => (
+        <View>
+          <Text className="text-red-400">{item.id}</Text>
+        </View>
+      ))}
+
+      {/* <Collapsible title="Light and dark mode components">
```
Comment (kekoawong) [inline app/(tabs)/explore.tsx:108]: Remove the unused code unless you are going to use it later

**Review by kekoawong — APPROVED** [review]
Just left minor comments for your discretion - looks good! Also for formatting - do we have a linter or prettier configured on this?

**Review by kekoawong — APPROVED** [review]

File: app/(tabs)/explore.tsx (hunk 1 above)
Comment (SethY8s) [reply app/(tabs)/explore.tsx:15]: yeah, without adding some other parts of the repo we can't really do that.

File: app/(tabs)/explore.tsx (hunk 3 above)
Comment (SethY8s) [reply app/(tabs)/explore.tsx]: removed it and added ESlint

**Pull request merged**